event_session_registration_multi_qty/models/event_session.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2017-19 Tecnativa - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl-3.0).
+from odoo import api, models
+
+
+class EventSession(models.Model):
+    _inherit = "event.session"
+
+    @api.depends("registration_ids.qty")
+    def _compute_seats(self):
+        for session in self:
+            if not session.event_id.registration_multi_qty:
+                return super()._compute_seats()
+            vals = {
+                "seats_unconfirmed": 0,
+                "seats_reserved": 0,
+                "seats_used": 0,
+                "seats_available": 0,
+                "seats_available_expected": 0,
+                "seats_available_pc": 0,
+            }
+            registrations = self.env["event.registration"].read_group(
+                [
+                    ("session_id", "=", session.id),
```

Comment:
Better to do this from outside the loop, to reduce the number of queries to compute this thing.
Otherwise you run a query per record, that's not efficient.

You probably need to use raw sql to achieve it, like it's done in the original method

Somthing like this perhaps

```py
    state_field = {
        'draft': 'seats_unconfirmed',
        'open': 'seats_reserved',
        'done': 'seats_used',
    }
    if self.ids:
        query = """
            SELECT session_id, state, sum(qty)
             FROM event_registration
             WHERE session_id IN %s
             AND state IN ('draft', 'open', 'done')
             GROUP BY session_id, state
        """
        self.env['event.registration'].flush(['session_id', 'state'])
        self.env.cr.execute(query, (tuple(self.ids),))
        res = self.env.cr.fetchall()
        for session_id, state, qty in res:
            results[session_id][state_field[state]] = num
```